lib/task/tools/dataIntegrityRepairTask.class.php

```diff
@@ -0,0 +1,359 @@
+        $task->run();
+
+        // Set root term as parent for terms without one
+        $sql = 'UPDATE term SET parent_id=110 WHERE parent_id IS NULL AND id<>110;';
+        $updated = QubitPdo::modify($sql);
+        $this->logSection('data-integrity-repair', sprintf("Updating terms without parent id: %d\n", $updated));
+
+        $this->log("Checking descriptions integrity:\n");
+
+        $sql = 'SELECT COUNT(io.id)
+          FROM information_object io
+          LEFT JOIN object o ON io.id=o.id
+          WHERE io.id<>1
+          AND o.id IS NULL;';
+        $this->log(sprintf("  - Descriptions without object row: %d\n", QubitPdo::fetchColumn($sql)));
+
+        $sql = 'SELECT COUNT(id)
+          FROM information_object
+          WHERE id<>1
+          AND parent_id IS NULL;';
+        $this->log(sprintf("  - Descriptions without parent id: %d\n", QubitPdo::fetchColumn($sql)));
+
+        $sql = 'SELECT COUNT(io.id)
+          FROM information_object io
+          LEFT JOIN information_object p ON io.parent_id=p.id
+          WHERE io.id<>1
+          AND p.id IS NULL;';
+        $this->log(sprintf("  - Descriptions without parent: %d\n", QubitPdo::fetchColumn($sql)));
+
+        $sql = 'SELECT COUNT(io.id)
+          FROM information_object io
+          LEFT JOIN status st ON io.id=st.object_id AND st.type_id=158
+          WHERE io.id<>1
+          AND st.status_id IS NULL;';
+        $this->log(sprintf("  - Descriptions without publication status: %d\n", QubitPdo::fetchColumn($sql)));
+
+        $sql = 'SELECT io.id, o.id as object_id, io.parent_id, p.id as parent, st.id as status, st.status_id
+          FROM information_object io
+          LEFT JOIN object o ON io.id=o.id
+          LEFT JOIN information_object p ON io.parent_id=p.id
+          LEFT JOIN status st ON io.id=st.object_id AND st.type_id=158
+          WHERE io.id<>1
+          AND (o.id IS NULL OR io.parent_id IS NULL
+          OR p.id IS NULL
+          OR st.id IS NULL
+          OR st.status_id IS NULL);';
+        $affectedIos = QubitPdo::fetchAll(
+        $sql, [], ['fetchMode' => PDO::FETCH_ASSOC]
+        );
+        $this->log(sprintf("  - Affected descriptions: %d\n", count($affectedIos)));
+
+        if (0 == count($affectedIos)) {
+            $this->log("All descriptions seem to be okay.\n");
+        } else {
+            $affectedIosAndDescendantIds = [];
+            $affectedIosById = [];
+            foreach (array_reverse($affectedIos) as $io) {
+                $this->populateAffectedIosAndDescendantIds($io['id'], $affectedIosAndDescendantIds);
+                $affectedIosById[$io['id']] = $io;
+            }
+            $this->logSection('data-integrity-repair', sprintf("  - Affected descriptions (including descendants): %d\n", count($affectedIosAndDescendantIds)));
+
+            switch ($options['fix']) {
+        case 'fix':
+          $this->report($filename, $affectedIosById, $affectedIosAndDescendantIds);
```

Comment:
Report happens in all cases, you could remove the breaks and make all cases fall down to the default, not needing to call the report generation on each case. You could also move the report generation before the switch and remove the default case.